### PR TITLE
Pass blockNumber to isValidServiceName() through isServAcntName()

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Pass blockNumber to isValidServiceName() through isServAcntName()
- Update base/genesis_rules.json but preserve genesis rules under mainnet-prod and testnet-prod
- Simplify some function parameters by using 'options' for timestamp, blockNumber, blockTime 
- Pass 'options' to chained operation execution and native functions calls

### CODE REVIEW

File: test/integration/function.test.js [hunk 1]
```diff
@@ -2302,6 +2312,51 @@ describe('Native Function', () => {
             server1 + `/get_value?ref=${transferFromBalancePath}`).body.toString('utf-8')).result;
         expect(fromAfterBalance).to.equal(fromBeforeBalance);
       });
+
+      it('transfer: transfer with invalid service account service name', async () => {
```
Comment (liayoo) [inline test/integration/function.test.js:2316]: 👍

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

File: db/functions.js [hunk 2]
```diff
@@ -148,21 +145,24 @@ class Functions {
   }
 
   triggerFunctions(
-      functionPath, pathVars, functionMap, valuePath, value, prevValue, auth, timestamp,
-      transaction, blockNumber, blockTime, blockchainParams) {
+      functionPath, pathVars, functionMap, valuePath, value, prevValue, auth, transaction,
+      blockchainParams, options) {
```
[the remark this answers is not in the document]
Comment (minsulee2) [reply db/functions.js:149]: 👍

**Review by minsulee2 — APPROVED** [review]
LGTM!

**Comment by platfowner** [comment]
Thanks for the swift review! now merging.. @liayoo @minsulee2

**Pull request merged**